File: routes/admin/catelist.js

var res = require('./lib/resp.js');
var cates = require("./models/cates.js");

if (!session.get("isAdmin")) {
  return res.redirect('/routes/admin/login.js');
}

var allData = cates.find();
var data = {
  tree: cates.getTree(allData),
};
res.render('./views/admin/catelist.ejs', data);

File: routes/admin/catelist.go
package admin

import "github.com/zengming00/go-testShop/lib"
import "github.com/zengming00/go-testShop/models"

func CateList(ctx *lib.HandlerContext) {
	var isAdmin = false
	if v, ok := ctx.GetSessionVal("isAdmin"); ok {
		isAdmin = v.(bool)
	}
	if !isAdmin {
		ctx.Redirect("/routes/admin/login.go")
	}

	var allData = models.Cates.Find()
	var data = map[string]interface{}{
		"tree": models.Cates.GetTree(allData),
	}
	ctx.Render("./views/admin/catelist.html", data, nil)
}
